ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoSignedTransaction.kt

```diff
@@ -0,0 +1,54 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import java.security.PublicKey
+
+/**
+ * Defines a signed UTXO transaction.
+ *
+ * @property id The ID of the transaction.
+ * @property signatures The signatures that have been applied to the transaction.
+ */
+@DoNotImplement
+@CordaSerializable
+interface UtxoSignedTransaction {
+
+    val id: SecureHash
+    val signatures: Set<DigitalSignatureAndMetadata>
+
+    /**
+     * Adds the specified signature to the current [UtxoSignedTransaction].
+     *
+     * @param signature The signature and metadata to add to the current [UtxoSignedTransaction].
+     * @return Returns a new [UtxoSignedTransaction] containing the applied signature.
+     */
+    fun addSignature(signature: DigitalSignatureAndMetadata): UtxoSignedTransaction
+
+    /**
+     * Adds the specified signatures to the current [UtxoSignedTransaction].
+     *
+     * @param signatures The signatures and metadata to add to the current [UtxoSignedTransaction].
+     * @return Returns a new [UtxoSignedTransaction] containing the applied signatures.
+     */
+    fun addSignatures(signatures: Iterable<DigitalSignatureAndMetadata>): UtxoSignedTransaction
```

Comment:
A vararg variant of this might be nice to have, and would probably make the single `addSignature` case redundant.

```kotlin
default fun addSignatures(vararg signatures: DigitalSignatureAndMetadata): UtxoSignedTransaction =
    addSignatures(signatures)
```